### DISCUSSION TITLE
How to provide Bluetooth LE functionality in both Mobile and Desktop?

### DISCUSSION THREAD

**Discussion by alessandrofrancesconi** [AvaloniaUI/Avalonia#10594] [Q&A]
First of all, sorry if my question might not seem strictly related to Avalonia. But... considering an Avalonia 11 multiplatform project created via `dotnet new avalonia.xplat`, so targeting Windows/MacOS/Android/iOS (not considering Web). How do I add Bluetooth LE functionalities (discover devices & transmit data)?

As of now I did not find any common namespaces in .NET Standard or .NET 7 frameworks. On Android and iOS, I could rely on libraries like [ble.net](https://github.com/nexussays/ble.net), but it is not compatible with Avalonia "Desktop".

Do you have some suggestions? Thank you.

**Comment by timunie** [comment]
I think you will need to add an interface / service which you implement for each OS and inject it into the main project.
